AdminWebsite/AdminWebsite/ClientApp/src/app/services/pipe-stringifier.service.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { TestBed } from '@angular/core/testing';
+
+import { PipeStringifierService } from './pipe-stringifier.service';
+
+describe('PipeStringifierService', () => {
+    let service: PipeStringifierService;
+    const testString = '|name|test|height|100';
+    const testObject = { name: 'test', height: '100' };
+    beforeEach(() => {
+        TestBed.configureTestingModule({});
+        service = TestBed.inject(PipeStringifierService);
+    });
+
+    it('should be created', () => {
+        expect(service).toBeTruthy();
+    });
+
+    it('should encode an object to a string', () => {
+        expect(service.encode(testObject)).toEqual(testString);
+    });
+
+    it('should decode a string into an object', () => {
+        expect(service.decode(testString)).toEqual(testObject);
```

Comment:
could we test a couple more not so happy paths?
`|name||height|100` // With empty
`|name|null|height|100` // With null
`|prop1|1|prop2|2|prop3|3` // 3 properties in values should parse to string
`|name|test|height|` // missing height
`|otherinfo|containingapipe|likethis`
null or empty